server/controllers/usercontroller.js

```diff
@@ -0,0 +1,113 @@
+import '@babel/plugin-transform-regenerator';
+import '@babel/polyfill';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import nodemailer from 'nodemailer';
+import userService from '../services/userService';
+
+dotenv.config();
+class User {
+  static async signUp(req, res) {
+    const getEmail = await userService.findByProp({ email: req.body.email });
+
+    if (getEmail[0]) {
+      return res.status(409).json({
+        status: 409,
+        error: 'This email already exist in the database',
+        path: 'email',
+      });
+    }
+    const newUser = {
+      firstname: req.body.firstname,
+      lastname: req.body.lastname,
+      email: req.body.email,
+      password: await bcrypt.hash(req.body.password, 10),
+    };
+
+    const inserter = await userService.createuser(newUser);
+    const userGot = inserter.dataValues;
+    const token = jwt.sign(
+      {
+        id: userGot.id,
+        email: userGot.email,
+      },
+      process.env.JWT_KEY,
+      {
+        expiresIn: '4h',
+      },
+    );
+
+    const url = `http://localhost:5000/verify/${token}`;
+    const transporter = nodemailer.createTransport({
```

Comment:
Can you make this in a separate file so that anyone who will send an email can use it by passing in params instead of creating a new one every time?

Reply:
Okay